### DISCUSSION TITLE
Support of custom fields in the browser extension

### DISCUSSION THREAD

**Discussion by TahirJalilov** [keepassxreboot/keepassxc#7311] [Ideas]
Make it possible to autodetect and fill custom fields by browser extension

**Comment by viharm** [comment]
Firstly, I'd like to thank the developes for such an awesome tool. I use this on both Linux and Windows desktops (with database sync using Seafile/WebDAV - since I sync to Android and iOS too). In addition to securing sensitive information, the deep integration with browsers and tools like ssh-agent and Pageant have truly changed my life - awesome!!!

I concur with @TahirJalilov . Some websites require logging in using email, but also have a username. In this instance, logging with the custom field instead of username would be ideal. This is supported in auto-type, but I couldn't find this in the Browser extension.

One work-around would be to put the email value in the username field, but this will require a separate custom field for the username value - a bit counter-intuitive and certainly a workaround. Moreover it would be very difficult for those with hundres or thousands of entries. E.g., I have just over 1100 entries in my database.

This change would really really help :-)

I'm happy to sponsor this change, if it is possible.

**Reply by droidmonkey** [reply]
It is supported by poorly documented. See here for instructions: https://keepassxc.org/docs/#faq-browser-string-fields

**Reply by viharm** [reply]
Thanks, this worked. I’m still a bit confused about how the ordering of these custom fields works, but I’ll try that on a more complex website.

Thanks again.

**Reply by gunjubas** [reply]
@droidmonkey Thanks a lot! You save me time.
